### ISSUE TITLE
Convert-BcAppsToRuntimePackages: allow to set showmycode, keepContainer, include sign app parameters

### ISSUE THREAD

**Issue by ti-jalopez** [microsoft/navcontainerhelper#2457]
**Sugestions to Convert-BcAppsToRuntimePackages**
I'm using Convert-BcAppsToRuntimePackages to create runtime packages, but I can do the next processes:
- Force ShowMyCode $false when download runtime package.
- Sign converted apps after convert to runtime package.

**Add param -ShowMyCode and -IncludeSourceInPackageFile**
Current version of Convert-BcAppsToRuntimePackages, in Get-NavAppRuntimePackage the parameter -ShowMyCode is not present:

- I suggest to add -ShowMyCode and -IncludeSourceInPackageFile params to Convert-BcAppsToRuntimePackages, to override the app.json settings.

**Add param -keepContainer and/or -pfxFile and -pfxPassword**
To sign app I've tested to use Sign-BcContainerApp after finish Convert-BcAppsToRuntimePackages, but the container is deleted. To solve it I propose two options:
- Add keepContainer parameter: in my script I can use -keepContainer to convert the app and after that sign it and delete de container.
- Or, add parameters to sign runtime apps: pfxFile and pfxPassword.

**Comment by freddydk** [comment]
I will add -ShowMyCode and -IncludeSourceInPackageFile and directly transport the values to the underlying function.

The code signing certificate should be handled by adding this parameter to the Convert-BcAppsToRuntimePackages:

```
    -afterEachRuntimeCreation { Param($ht) 
        Write-Host "Signing runtime package"
        Sign-BcContainerApp -containerName bctest -appFile $ht.runtimeFile -pfxFile ($CodeSignCertificateUrlSecret.SecretValue | Get-PlainText) -pfxPassword $CodeSignCertificatePasswordSecret.SecretValue
    }
```

That works today.
Basically, that scriptblock is executed for every app and the appFile and the runtimeFile are in the hashtable transferred as parameter.

**Comment by ti-jalopez** [comment]
> he code signing certificate should be handled by adding this parameter to the Convert-BcAppsToRuntimePackages:
> 
> ```
>     -afterEachRuntimeCreation { Param($ht) 
>         Write-Host "Signing runtime package"
>         Sign-BcContainerApp -containerName bctest -appFile $ht.runtimeFile -pfxFile ($CodeSignCertificateUrlSecret.SecretValue | Get-PlainText) -pfxPassword $CodeSignCertificatePasswordSecret.SecretValue
>     }
> ```

I've modified my code to add an adaptated version of your proposal in afterEachRuntimeCreation, and I think it runs fine !!!

```
2022-05-18T06:36:47.0818030Z Publishing MyCompany_MyApp_0.4.19001.110.app
2022-05-18T06:37:01.0636790Z Creating Runtime Package MyCompany_MyApp_0.4.19001.110.runtime-20.1.39764.39901-es.app
2022-05-18T06:37:05.1053810Z Signing runtime package
2022-05-18T06:37:05.3609377Z Copying certificate file to container
2022-05-18T06:37:05.7370187Z Downloading Signing Tools
2022-05-18T06:37:06.0323880Z Installing Signing Tools
2022-05-18T06:37:14.9344300Z Signing c:\dest\MyCompany_MyApp_0.4.19001.110.runtime-20.1.39764.39901-es.app
2022-05-18T06:37:15.6228425Z Done Adding Additional Store
2022-05-18T06:37:16.4437056Z Successfully signed: c:\dest\MyCompany_MyApp_0.4.19001.110.runtime-20.1.39764.39901-es.app
2022-05-18T06:37:16.4438943Z 
```
Thank you very much :=)

**Comment by ti-jalopez** [comment]
> I will add -ShowMyCode and -IncludeSourceInPackageFile and directly transport the values to the underlying function.

I think I can't force to disable the parameter as the actual implementation using [switch] params.
Samples to force NO show code:
-ShowMyCode:$false
-IncludeSourceInPackageFile:$false

Could you change a bit the new funciontality?

Best regards,

**Comment by ti-jalopez** [comment]
> > I will add -ShowMyCode and -IncludeSourceInPackageFile and directly transport the values to the underlying function.
> 
> I think I can't force to disable the parameter as the actual implementation using [switch] params. Samples to force NO show code: -ShowMyCode:$false -IncludeSourceInPackageFile:$false
> 
> Could you change a bit the new funciontality?
> 
> Best regards,

Hi @freddydk ,
Could you change the behavior of new params ShowMyCode and IncludeSourceInPackage to make posiblle pass to downlad runtimepackage as value $false.
In your implementation, I can't do that, and if no params in download runtime, by defaut use de showmycode of app.json.

Best regards,
José Antonio López

**Comment by freddydk** [comment]
Will do

**Comment by freddydk** [comment]
I accidently rolled out 3.0.9 without this fix.
I have already rolled out a 3.0.10 preview with this fix - could you please test this, then I will roll out 3.0.10 as soon as I have confirmation that this works - as this is a breaking change from 3.0.9 (I don't want a lot of people using the switch parameters in 3.0.9).

Thanks and sorry.

**Comment by freddydk** [comment]
Amazed by the number of downloads of a new version...:-)
![image](https://user-images.githubusercontent.com/10775043/169974150-b564749b-9d37-4f8f-9d47-1ec812d1ce77.png)
Fortunately, they don't all write new scripts with this function...

**Comment by ti-jalopez** [comment]
> I accidently rolled out 3.0.9 without this fix. I have already rolled out a 3.0.10 preview with this fix - could you please test this, then I will roll out 3.0.10 as soon as I have confirmation that this works - as this is a breaking change from 3.0.9 (I don't want a lot of people using the switch parameters in 3.0.9).
> 
> Thanks and sorry.

Hi @freddydk ,
I'll do this week. I need to create an environment for upgrade de preview version, because I won't to upgrade preview version on production environment.
Best regards,

**Comment by freddydk** [comment]
Shipped in 3.0.10

Both parameters are now booleans and only added to the underlying cmdlet if specified in the first place.
This means that the function signature is stable.

**Comment by ti-jalopez** [comment]
> I will test the command today and perform the release - then you can try this out on your normal pipelines.

I've done some test with new version 3.0.10, and initially all runs fine 👍 

Thanks for all your great job!!

Best regards,
